### PR TITLE
Using dlcdn.apache.org for apache archive downloading 

### PR DESCRIPTION
**Pull request by turboFei**
### Why are the changes needed?

GA failed due to network unreachable.
```
Nov 25, 2024 5:10:56 AM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.net.SocketException) caught when processing request to {s}->[https://archive.apache.org:443:](https://archive.apache.org/) Network is unreachable
Nov 25, 2024 5:10:56 AM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {s}->[https://archive.apache.org:443](https://archive.apache.org/)
Nov 25, 2024 5:11:56 AM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.net.SocketException) caught when processing request to {s}->[https://archive.apache.org:443:](https://archive.apache.org/) Network is unreachable
Nov 25, 2024 5:11:56 AM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {s}->[https://archive.apache.org:443](https://archive.apache.org/)
Nov 25, 2024 5:12:56 AM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.net.SocketException) caught when processing request to {s}->[https://archive.apache.org:443:](https://archive.apache.org/) Network is unreachable
Nov 25, 2024 5:12:56 AM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {s}->[https://archive.apache.org:443](https://archive.apache.org/)
Warning:  The requested profile "gen-policy" could not be activated because it does not exist.
Error:  Failed to execute goal com.googlecode.maven-download-plugin:download-maven-plugin:1.11.3:wget (download-spark-release) on project kyuubi-download: Network is unreachable -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.googlecode.maven-download-plugin:download-maven-plugin:1.11.3:wget (download-spark-release) on project kyuubi-download: Network is unreachable
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
```

### How was this patch tested?

GA.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -125,7 +125,7 @@
         <!-- Please don't upgrade the version to 4.10+, it depends on JDK 11 -->
         <antlr4.version>4.9.3</antlr4.version>
         <antlr.st4.version>4.3.4</antlr.st4.version>
-        <apache.archive.dist>https://archive.apache.org/dist</apache.archive.dist>
+        <apache.archive.dist>https://dlcdn.apache.org</apache.archive.dist>
```
Comment (pan3793) [inline pom.xml:128]: does this work for archived versions?

Comment (turboFei) [reply pom.xml:128]: image

Not sure whether the network is reachable, need to test.

Comment (turboFei) [reply pom.xml:128]: ok, there is no `archived` version ...

Comment (pan3793) [reply pom.xml:128]: BTW, the tgz should be loaded from the cache directly, unless the recent plugin bumping https://github.com/apache/kyuubi/pull/6807 causes some issues

Comment (turboFei) [reply pom.xml:128]: Thanks for the info.

**Pull request closed without merging**